src/odemis/driver/xt_client.py

```diff
@@ -1070,6 +1135,38 @@ def _updateSettings(self):
                 self.magnification._value = mag
                 self.horizontalFoV.notify(fov)
                 self.magnification.notify(mag)
+            pitch = self.parent.get_pitch()
+            if pitch != self.pitch.value:
+                self.pitch._value = pitch
+                self.pitch.notify(pitch)
+            beam_stigmator = self.parent.get_primary_stigmator()
+            if beam_stigmator != self.beamStigmator.value:
+                self.beamStigmator._value = beam_stigmator
+                self.beamStigmator.notify(beam_stigmator)
+            pattern_stigmator = self.parent.get_secondary_stigmator()
+            if pattern_stigmator != self.patternStigmator.value:
+                self.patternStigmator._value = pattern_stigmator
+                self.patternStigmator.notify(pattern_stigmator)
+            beam_shift_transformtion_matrix = self.parent.get_dc_coils()
+            if beam_shift_transformtion_matrix != self.beamShiftTransformtionMatrix.value:
+                self.beamShiftTransformtionMatrix._value = beam_shift_transformtion_matrix
+                self.beamShiftTransformtionMatrix.notify(beam_shift_transformtion_matrix)
+            mpp_rotation = self.parent.get_mpp_orientation()
+            if mpp_rotation != self.multiprobeRotation.value:
+                self.multiprobeRotation._value = mpp_rotation
+                self.multiprobeRotation.notify(mpp_rotation)
+            aperture_index = self.parent.get_aperture_index()
+            if aperture_index != self.apertureIndex.value:
+                self.apertureIndex._value = aperture_index
+                self.apertureIndex.notify(aperture_index)
+            beamlet_index = self.parent.get_beamlet_index()
+            if beamlet_index != self.beamletIndex.value:
+                self.beamletIndex._value = tuple(int(i) for i in beamlet_index)
```

Comment:
Could you convert to a tuple of ints *before* the if? Assuming the float is precisely equivalent to the int, it'd work fine as is, but if for some reason it's a tiny bit smaller or bigger (due to some floating error), you would end up updating the beamletIndex every time.